core/lib/zksync_core/src/eth_sender/tests.rs

```diff
@@ -657,59 +703,72 @@ async fn correct_order_for_confirmations() -> anyhow::Result<()> {
 }
 
 #[tokio::test]
-async fn skipped_l1_batch_at_the_start() -> anyhow::Result<()> {
-    let connection_pool = ConnectionPool::test_pool().await;
-    let l1_batch_commit_data_generator = Arc::new(RollupModeL1BatchCommitDataGenerator {});
-    let mut tester = EthSenderTester::new(
-        connection_pool,
+async fn correct_order_for_confirmations() -> anyhow::Result<()> {
+    let mut rollup_tester = EthSenderTester::new(
+        ConnectionPool::test_pool().await,
         vec![100; 100],
-        true,
-        l1_batch_commit_data_generator.clone(),
+        false,
+        Arc::new(RollupModeL1BatchCommitDataGenerator {}),
     )
     .await;
+    let mut validium_tester = EthSenderTester::new(
+        ConnectionPool::test_pool().await,
+        vec![100; 100],
+        false,
+        Arc::new(ValidiumModeL1BatchCommitDataGenerator {}),
+    )
+    .await;
+
+    _correct_order_for_confirmations(&mut rollup_tester).await?;
+    _correct_order_for_confirmations(&mut validium_tester).await
+}
+
+async fn _skipped_l1_batch_at_the_start(tester: &mut EthSenderTester) -> anyhow::Result<()> {
+    let l1_batch_commit_data_generator: Arc<dyn L1BatchCommitDataGenerator> =
+        tester.aggregator.get_l1_batch_commit_data_generator();
```

Comment:
Can't we just do this? (We'd get rid of the getter method)
```suggestion
    let l1_batch_commit_data_generator: Arc<dyn L1BatchCommitDataGenerator> =
        tester.aggregator.l1_batch_commit_data_generator.clone();
```

Reply:
In order to do that, we should expose public the aggregator field inside the `EthTxAggregator` and the field `l1_batch_commit_data_generator`.